Rover/mode_circle.cpp

```diff
@@ -190,12 +201,16 @@ void ModeCircle::update_circling()
     const float accel_fb = constrain_float(config.speed - target.speed, -speed_change_max, speed_change_max);
     target.speed += accel_fb;
 
-    // calculate angular rate and update target angle
-    const float circumference = 2.0 * M_PI * config.radius;
-    const float angular_rate_rad = (target.speed / circumference) * M_2PI * (config.dir == Direction::CW ? 1.0 : -1.0);
-    const float angle_dt = angular_rate_rad * rover.G_Dt;
-    target.yaw_rad = wrap_PI(target.yaw_rad + angle_dt);
-    angle_total_rad += angle_dt;
+    // calculate angular rate and update target angle, if the vehicle is not trying to track back
+    if (!tracking_back) {
+        const float circumference = 2.0 * M_PI * config.radius;
+        const float angular_rate_rad = (target.speed / circumference) * M_2PI * (config.dir == Direction::CW ? 1.0 : -1.0);
+        const float angle_dt = angular_rate_rad * rover.G_Dt;
+        target.yaw_rad = wrap_PI(target.yaw_rad + angle_dt);
+        angle_total_rad += angle_dt;
+    } else {
+        init_target_yaw_rad();
```

Comment:
This means that we effectively reset the target position to the closest position on the circle. So the position controller will always be pushing radially. But there is still some target velocity and acceleration. Because only the target position is changed were not giving kinematically consistent targets. You could swap to just giving a position target like we do on the `drive_to_radius` stage. You could just kick back into that stage and reuse the `reached_edge` flag.

Reply:
My understanding is that the target velocity/accel are calculated further down in ``update_circling()`` based on the target yaw, keeping it consistent.

A position target won't work because we need to vehicle at the correct heading (velocity) to have a smooth entry into the circle:
![Screenshot from 2024-09-09 20-41-01](https://github.com/user-attachments/assets/a8a6641a-d291-40a7-8229-25f8950913cb)

Reply:
This is what the PR currently does. I think it's a smoother approach back into the circle:
![Screenshot from 2024-09-09 20-52-42](https://github.com/user-attachments/assets/4c01dc86-0eea-49ea-b081-f25710c8cf5f)

Reply:
OK, i'm convinced.